modules/KalturaSupport/components/chapters/chapters.js

```diff
@@ -463,7 +463,10 @@
 			//Get current item number
 			var orderId = this.mediaList.length;
 			//Map items to mediaList items
-			var mediaItems = $.map(items, function (item) {
+			var filterItems = items.filter(function( item ) {
+				return _this.getPlayer().kCuePoints.validateCuePointTags(item, _this.getPlayer().kCuePoints.getPreviewCuePointTag());
```

Comment:
@MaksymKorniiov 

With this implementation we are calling _this.getPlayer().kCuePoints.getPreviewCuePointTag() every filter iteration. We should do it once before the filter